assets/src/admin/paired-browsing/app.js

```diff
@@ -227,108 +366,112 @@ class PairedBrowsingApp {
 	}
 
 	/**
-	 * Checks if a URL has the 'amp_validation_errors' query variable.
+	 * Replace location.
 	 *
-	 * @param {string} url URL string.
-	 * @return {boolean} True if such query var exists, false if not.
+	 * @param {HTMLIFrameElement} iframe IFrame Element.
+	 * @param {string}            url    URL.
+	 */
+	replaceLocation( iframe, url ) {
+		this.sendMessage(
+			iframe.contentWindow,
+			'replaceLocation',
+			{ href: url },
+		);
+	}
+
+	/**
+	 * Receive scroll.
+	 *
+	 * @param {Object} data         Data.
+	 * @param {number} data.x       X position.
+	 * @param {number} data.y       Y position.
+	 * @param {Window} sourceWindow The source window.
 	 */
-	urlHasValidationErrorQueryVar( url ) {
-		return hasQueryArg( url, 'amp_validation_errors' );
+	receiveScroll( { x, y }, sourceWindow ) {
+		// Rely on scroll event to determine initially-active iframe before mouse first moves.
+		if ( ! this.activeIframe ) {
+			this.activeIframe = this.isAmpWindow( sourceWindow )
+				? this.ampIframe
+				: this.nonAmpIframe;
+		}
+
+		// Ignore scroll events from the non-active iframe.
+		if ( ! this.activeIframe || sourceWindow !== this.activeIframe.contentWindow ) {
+			return;
+		}
+
+		const otherWindow = this.isAmpWindow( sourceWindow )
+			? this.nonAmpIframe.contentWindow
+			: this.ampIframe.contentWindow;
+		this.sendMessage( otherWindow, 'scroll', { x, y } );
 	}
 
 	/**
-	 * Registers the provided client window with its parent, so that it can be managed by it.
+	 * Receive loaded.
 	 *
-	 * @param {Window} win Document window.
+	 * @param {Object}  data                   Data.
+	 * @param {boolean} data.isAmpDocument     Whether the document is actually an AMP page.
+	 * @param {?string} data.ampUrl            The AMP URL.
+	 * @param {?string} data.nonAmpUrl         The non-AMP URL.
+	 * @param {string}  data.documentTitle The canonical link URL if present.
+	 * @param {Window}  sourceWindow The source window.
```

Comment:
It seems `data.ampUrl` and `data.nonAmpUrl` are required, so I'd suggest removing the `?` in the param type. Also, it seems the description for `documentTitle` is incorrect, so I'm suggesting a correct one.

```suggestion
	 * @param {Object}  data                Data.
	 * @param {boolean} data.isAmpDocument  Whether the document is actually an AMP page.
	 * @param {string} data.ampUrl         The AMP URL.
	 * @param {string} data.nonAmpUrl      The non-AMP URL.
	 * @param {string}  data.documentTitle  The title of the document.
	 * @param {Window}  sourceWindow        The source window.
```